keepsorted/line_group.go

```diff
@@ -57,14 +57,46 @@ type lineGroupContent struct {
 }
 
 type lineGroupCalculations struct {
-	commentOnly *bool
-	regexes     []regexToken
-	lines       string
-	comment     string
+	commentOnly   *bool
+	regexes       []regexToken
+	internalLines string
+	lines         string
+	comment       string
 }
 
 type regexToken []*captureGroupToken
 
+func (t regexToken) wasUsed() bool {
+	if t == nil {
+		// Report that the regex didn't match.
+		return true
+	}
+	for _, cg := range t {
+		if cg.wasUsed() {
+			return true
+		}
+	}
+	return false
+}
+
+func (t regexToken) GoString() string {
```

Comment:
A side note that normally GoString is used to represent things in a format that would be valid Go code. I wonder if we should just implement String() instead, but I don't care that much and that's merely a convention and doesn't have to be followed.

Reply:
Hmm that's fair. We use `String` on `LineGroup` to convert it back to its original form, so I was just using `GoString` as a way to print debug information. I can switch to `DebugString` to make that explicit